File: 1089 - Duplicate Zeros.js
/**
 * @param {number[]} arr
 * @return {void} Do not return anything, modify arr in-place instead.
 */
var duplicateZeros = function (arr) {
    let q = [];
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] == 0) {
            q.push(0, 0);
        } else {
            q.push(arr[i]);
        }
        arr[i] = q.shift();
    }
};

File: 1089 - Duplicate Zeros.py
class Solution:
    def duplicateZeros(self, arr: List[int]) -> None:
        """
        Do not return anything, modify arr in-place instead.
        """
        dup = 0
        q = []
        for i in range(len(arr)):
            if arr[i] == 0:
                q.append(0)
                q.append(0)
            else:
                q.append(arr[i])
            arr[i] = q.pop(0)
